ovs/extensions/healthcheck/openvstorage/openvstoragecluster_health_check.py

```diff
@@ -0,0 +1,970 @@
+            voldrv_volume_list = voldrv_client.list_volumes()
+
+            # collect data from model
+            model_vdisk_list = vp.vdisks
+
+            # crossreference model vs. volumedriver
+            for vdisk in model_vdisk_list:
+                if not vdisk.volume_id in voldrv_volume_list:
+                    missinginvolumedriver.append(vdisk.volume_id)
+
+            # crossreference volumedriver vs. model
+            # (This can be a performance bottleneck on heavy env. due to nested for loops)
+            for volume in voldrv_volume_list:
+                for vdisk in model_vdisk_list:
+                    if str(volume) != vdisk.volume_id and len(model_vdisk_list) == (model_vdisk_list.index(vdisk)+1):
+                        missinginmodel.append(volume)
+
+            # display discrepancies for vPool
+            if len(missinginvolumedriver) != 0:
+                self.LOGGER.logger("Detected volumes that are MISSING in volumedriver but ARE in ovsdb in vPool "
+                                   "'{0}': {1}".format(vp.name, ', '.join(missinginvolumedriver)), self.module, 0,
+                                   'discrepancies_ovsdb_{0}'.format(vp.name))
+            else:
+                self.LOGGER.logger("NO discrepancies found for ovsdb in vPool '{0}'".format(vp.name), self.module, 1,
+                                   'discrepancies_ovsdb_{0}'.format(vp.name))
+
+            if len(missinginmodel) != 0:
+                self.LOGGER.logger("Detected volumes that are AVAILABLE in volumedriver but ARE NOT in ovsdb in vPool "
+                                   "'{0}': {1}".format(vp.name, ', '.join(missinginmodel)), self.module, 0,
+                                   'discrepancies_voldrv_{0}'.format(vp.name))
+            else:
+                self.LOGGER.logger("NO discrepancies found for voldrv in vPool '{0}'".format(vp.name), self.module, 1,
+                                   'discrepancies_voldrv_{0}'.format(vp.name))
+
+    def check_for_halted_volumes(self):
+        """
+        Checks for halted volumes on a single or multiple vPools
+        """
+
+        self.LOGGER.logger("Checking for halted volumes: ", self.module, 3, 'checkHaltedVolumes', False)
+
+        vpools = VPoolList.get_vpools()
+
+        if len(vpools) != 0:
+
+            for vp in vpools:
+
+                haltedvolumes = []
+
+                self.LOGGER.logger("Checking vPool '{0}': ".format(vp.name), self.module, 3,
+                                   'checkVPOOL_{0}'.format(vp.name), False)
+
+                config_file = self.utility.get_config_file_path(vp.name, self.machine_id, 1, vp.guid)
+                voldrv_client = src.LocalStorageRouterClient(config_file)
+
+                for volume in voldrv_client.list_volumes():
+                    # check if volume is halted, returns: 0 or 1
+                    if int(self.utility.convert_xml_to_json(voldrv_client.info_volume(volume))["boost_serialization"]
```

Comment:
```voldrv_client.info_volume(volume)``` returns an object so you can just use plain properties. No need to convert it to a string, then parse the xml and convert it to a json to then convert it to a dict.